deviceModeInit.js

```diff
@@ -153,32 +159,46 @@ var rudderTracking = (function () {
 
   function checkCartAttributes(cart) {
     const { attributes } = cart;
-    if (attributes?.rudderAnonymousId && attributes?.rudderUpdatedAt) {
-      return true;
+    if (attributes?.rudderAnonymousId) {
+      return false;
     }
-    return false;
+    return true;
   }
 
+  function updateCookies() {
+    const expiryInHours = 2;
+    const date = new Date();
+    date.setTime(date.getTime() + (expiryInHours * 60 * 60 * 1000));
+    const expires = "expires=" + date.toUTCString();
+    document.cookie = `rs_shopify_updated_at = ${Date.now()}; ${expires};`;
+  }
   function updateCartAttribute() {
     const anonymousId = rudderanalytics.getAnonymousId();
     return rs$.post(
       window.Shopify.routes.root + "cart/update.json",
       {
         attributes: {
           rudderAnonymousId: anonymousId,
-          rudderUpdatedAt: Date.now(),
         },
       },
       undefined,
       "json"
     );
   }
-
-  function checkUpdateTime(cart) {
-    const oneHourTimeInMilliSeconds = 60 * 60 * 1000;
-    const { rudderUpdatedAt } = cart.attributes;
-    const currentTime = Date.now();
-    return currentTime - rudderUpdatedAt > oneHourTimeInMilliSeconds;
+  function checkUpdateTime() {
+    const allcookies = document.cookie;
+    const cookiearray = allcookies.split(";");
+    cookiearray.forEach((cookie) => {
+      const keyVal = cookie.split("=");
+      const name = keyVal[0];
+      const value = keyVal[1];
+      if (name === "rs_shopify_updated_at") {
+        const oneHourTimeInMilliSeconds = 60 * 60 * 1000;
+        const currentTime = Date.now();
+        return currentTime - value > oneHourTimeInMilliSeconds;
+      }
+    });
+    return true;
```

Comment:
There is already a cookie utility below in you can use that

Reply:
Done